From 2be9f812d1c76191204a9970034d73371eda9cd0 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Tue, 21 Mar 2017 17:22:20 +0800
Subject: [PATCH 1596/5242] MLK-11665 tty: serial: imx: move dma init to
 .startup() function

commit  322e372bc67bb62b49556c9cc2df6b5e6a9ec95c from
https://source.codeaurora.org/external/imx/linux-imx.git

Kernel upgrade to 4.1.8 from 4.1.4 has one conflict: commit:3cd6a7db4c2c and
commit:4eede03b97bf, and there introduces one issue during the merge. The issue
cause pio mode cannot work.

The patch fix the error that move the dma init function to .startup().

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index f9e9eb9..4451d3d 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1404,6 +1404,11 @@ static int imx_uart_startup(struct uart_port *port)
 	while (!(imx_uart_readl(sport, UCR2) & UCR2_SRST) && (--i > 0))
 		udelay(1);
 
+	/* Can we enable the DMA support? */
+	if (is_imx6q_uart(sport) && !uart_console(port)
+		&& !sport->dma_is_inited)
+		imx_uart_dma_init(sport);
+
 	spin_lock_irqsave(&sport->port.lock, flags);
 
 	/*
@@ -1630,11 +1635,6 @@ static void imx_uart_flush_buffer(struct uart_port *port)
 			} else {
 				imx_uart_rts_auto(sport, &ucr2);
 			}
-
-			/* Can we enable the DMA support? */
-			if (imx_uart_is_imx6q(sport) && !uart_console(port)
-				&& !sport->dma_is_inited)
-				imx_uart_dma_init(sport);
 		} else {
 			termios->c_cflag &= ~CRTSCTS;
 		}
-- 
1.7.9.5

